include/finalfusion.h

```diff
@@ -26,6 +27,21 @@ typedef struct ff_embeddings_t *ff_embeddings;
    */
   void ff_free_embeddings(ff_embeddings embeddings);
 
+  /**
+   * Returns the dimensionality of the embeddings.
+   */
+  uintptr_t dims(ff_embeddings embeddings);
+
+  /**
+   * Look up the embedding for the query word and write to buffer.
+   *
+   * Callers need to verify that the buffer is large enough for the
+   * dimensionality of the embeddings.
+   *
+   * Returns 1 if an embedding could be found, 0 otherwise.
+   */
+  uintptr_t embedding(ff_embeddings embeddings, const char* word, float* buffer);
```

Comment:
Use `char const *word`. I know that not everyone does this, but it is good style, because then the rule 'the type that is left of the `const` is constant applies'. See

http://www.icce.rug.nl/documents/cplusplus/cplusplus03.html#l31

for a more extensive motivation.

Reply:
Should return `size_t` or `int` (and change the corresponding Rust signature). `int` is sort of the standard for returning error codes in C. If we only support C99 upwards, we could also use `bool`. Though I am fine with `size_t` as well.

Reply:
Changed the return type to int